app/models/discovery_rule.rb

```diff
@@ -16,6 +16,7 @@ class DiscoveryRule < ApplicationRecord
   validates_lengths_from_database
   before_validation :default_int_attributes
   before_validation :enforce_taxonomy
+  before_validation :hostgroup_present
```

Comment:
Why not `validates_presence_of :hostgroup`?

Reply:
@timogoebel hey, `validates_presence_of` just validates that the specified attributes are not blank (as defined by Object#blank?). Here, i am trying to make a call to the database and check whether the value is present or not? also making sure a `dummy` value does not pass through.

Reply:
@rahulbajaj0509: If I read the rails source code correctly, it just calls `hostgroup.blank?` internally. Isn't that the same thing you are doing?

Reply:
Well the standard practice is to use `validates_presence_of`. Can you provide an example when the standard approach would not work and yours would please?

Reply:
Be sure to use `validates_presence_of :hostgroup` and not `validates_presence_of :hostgroup_id`. The latter is wrong.

Reply:
@lzap 
CASE 1: If you use `validates_presence_of` and run: 
hammer -u admin -p changeme discovery-rule create --hostgroup-id 999 --locations="Default Location" --name "rabajaj" --search "last_report = Today" --organizations="Default Organization"

Rule will be created

CASE 2: If we use the function provided in this patch:
Rule will not get created.

Reply:
I'm wondering if we have this bug across the whole app.

Reply:
We do use foreign keys for that both in core and in discovery, please add it to this association and that should do it.

Reply:
Note you might want to create FK via migration, I don't remember how this is done but http://guides.rubyonrails.org/active_record_migrations.html#foreign-keys